##### 1. Lectura de archivos

In [1]:
# Importar pandas y numpy
import pandas as pd
import numpy as np

# Leer archivo csv
df = pd.read_csv('worldcup2014.csv')

# Solicitar 5 primeras observaciones
df.head()

team     continent group  group_pos  cantidad_juegos  \
0       Brazil  southamerica     A          1                3   
1       Mexico  northamerica     A          2                3   
2      Croatia        europe     A          3                3   
3     Cameroon        africa     A          4                3   
4  Netherlands        europe     B          1                3   

   juegos_ganados  juegos_empatados  juegos_perdidos  goles_favor  \
0               2                 1                0            7   
1               2                 1                0            4   
2               1                 0                2            6   
3               0                 0                3            1   
4               3                 0                0           10   

   goles_contra goles_diferencia  puntos  clasificado  
0             2                5       7            1  
1             1                3       7            1  
2             6                0       3            0  
3             9               −8       0            0  
4             3                7       9            1

##### 2. Frecuencia de continente

In [72]:
# Continente con una mayor presencia
counts = df['continent'].value_counts()[:1]
print(f'El continente con mayor presencia es:\n{counts}')

# Probabilidad de elegir un equipo asiatico al azar
p_asia = df['continent'].value_counts('%')['asia']
print(f'Probabilidad de elegir un equipo asiatico: {p_asia*100}%')

# Probabilidad de elegir un equipo africano al azar
p_africa = df['continent'].value_counts('%')['africa']
print(f'Probabilidad de elegir un equipo africano: {p_africa*100}%')

El continente con mayor presencia es:
europe    13
Name: continent, dtype: int64
Probabilidad de elegir un equipo asiatico: 12.5%
Probabilidad de elegir un equipo africano: 15.625%


- El continente con mayor presencia es Europa con 13 repeticiones.
- La probabilidad de elegir un equipo asiatico es de 12,5%
- La probabilidad de elegir un equipo africano es de 15,6%

##### 3. Generación de un subconjunto de datos

In [73]:
europe =        df[df['continent'] == 'europe'].reset_index(drop=True)
southamerica =  df[df['continent'] == 'southamerica'].reset_index(drop=True)
africa =        df[df['continent'] == 'africa'].reset_index(drop=True)
northamerica =  df[df['continent'] == 'northamerica'].reset_index(drop=True)
asia =          df[df['continent'] == 'asia'].reset_index(drop=True)

##### 4. Probabilidad de clasificación a la siguiente ronda

In [75]:
df2 = df[['continent','clasificado']].groupby('continent',as_index=False).agg(['sum','count'])
df2 = df2['clasificado'].reset_index()
df2['prob'] = df2['sum'] / df2['count']
df2

continent  sum  count      prob
0        africa    2      5  0.400000
1          asia    0      4  0.000000
2        europe    6     13  0.461538
3  northamerica    3      4  0.750000
4  southamerica    5      6  0.833333

- La probabilidad que clasifique un equipo asiatico a la segunda ronda es 0%
- Los 2 continentes con mayor probabilidad de clasificar son Sudamerica con 83% y Norteamericanos con 75%
- La probabilidad de que un país europeo no clasifique es de 53,8%

##### 5. Probabilidad de victoria y derrota

In [106]:
df['ha_ganado'] = 0
df.loc[df['juegos_ganados'] > 0, 'ha_ganado'] = 1

In [105]:
df['ha_perdido'] = 0
df.loc[df['juegos_perdidos'] > 0, 'ha_perdido'] = 1

In [103]:
df2 = df[['continent','ha_ganado']].groupby(['continent'],as_index=False).sum('ha_ganado').sort_values('ha_ganado',ascending=False) #agg(['sum','count'])
#df2 = df2['ha_ganado'].reset_index()
df2['prob'] = df2['ha_ganado'] / len(df)
df2
#df['ha_ganado'] = 

continent  ha_ganado     prob
2        europe         11  0.34375
4  southamerica          6  0.18750
0        africa          3  0.09375
3  northamerica          3  0.09375
1          asia          0  0.00000

In [104]:
df3 = df[['continent','ha_perdido']].groupby(['continent'],as_index=False).sum('ha_perdido').sort_values('ha_perdido',ascending=False) #agg(['sum','count'])
#df2 = df2['ha_ganado'].reset_index()
df3['prob'] = df3['ha_perdido'] / len(df)
df3

continent  ha_perdido     prob
2        europe           9  0.28125
0        africa           5  0.15625
1          asia           4  0.12500
4  southamerica           3  0.09375
3  northamerica           2  0.06250

- A. Europa tiene la mayor probabilidad de tener paises con al menos una victoria con un 34.3%.
- B. Africa y Norteamérica tienen la misma probabilidad de tener paises con al menos una victoria con un 9.4%. Mientras que, los continentes con mayor probabilidad de tener paises sin juegos perdidos son Europa y África con un 28.1% y un 15.6% respectivamente.